components/analytics/team_analytics/team_analytics.tsx

```diff
@@ -25,20 +27,21 @@ import { formatPostsPerDayData, formatUsersWithPostsPerDayData } from '../format
 const LAST_ANALYTICS_TEAM = 'last_analytics_team';
 
 //TODO find out where the types for these are initialized if you do end up using these
-// type Props = {
-//     actions: any;
-//     teams: object[];
-//     stats: object;
-//     locale: string;
-// }
-
-// type State = {
-//     team: any;
-//     recentlyActiveUsers: [];
-//     newUsers: [];
-// }
-
-export default class TeamAnalytics extends React.PureComponent {  //? do I have to add the <P,S> parameters here?
+type Props = {
+    actions: any;
+    teams: Team[];
+    stats: RelationOneToOne<Team, TeamStats>;
+    locale: string;
+}
+
+type State = {
+    team: Team | null;
+    recentlyActiveUsers: [];
+    newUsers: [];
+    id: string;
+}
+
+export default class TeamAnalytics extends React.PureComponent<Props, State> {  //? do I have to add the <P,S> parameters here?
```

Comment:
I've made a couple changes after finding the docs on converting the types, but there's still some ambiguity about some of the types and how the function arguments should be formatted.  I referenced another file: `needs_team.tsx` but there aren't some one-to-one conversions that are obvious to me.

Reply:
Yes, the generics you've used here are correct.